### PR TITLE
Further refactoring for the geo coding utility

### PR DESCRIPTION
**Pull request by dshemetov**
The work for this #215

### CODE REVIEW

**Review by chinandrew — APPROVED** [review]
Tests pass for me now. Linter yelled at some lines >100char, though not sure if we're using 100 or 120... otherwise looks good.

**Comment by dshemetov** [comment]
@chinandrew I think those are the long URL lines. Is there are a preferred way to handle those?

**Comment by chinandrew** [comment]
> @chinandrew I think those are the long URL lines. Is there are a preferred way to handle those?

if its a URL it's probably fine to leave it. might be able to do backslashes but that might actually make things less readable. Some of thse look like normal dosctrings though

Screenshot from 2020-09-30 14-24-57

**Comment by dshemetov** [comment]
Oh oops, thanks for catching those. Just fixed those up.

**Comment by dshemetov** [comment]
Just realized there are some renamings in this refactor that need to be smoothed out with the previous extension of the utility to JHU. Should be just naming fixes.

File: _delphi_utils_python/delphi_utils/geomap.py [hunk 1]
```diff
@@ -1,825 +1,1252 @@
...
+        state_codes = ["state_code", "state_id", "state_name"]
+
+        if not is_string_dtype(df[from_col]):
+            if from_code in ["fips", "zip"]:
+                df[from_col] = df[from_col].astype(str).str.zfill(5)
+            else:
+                df[from_col] = df[from_col].astype(str)
+
+        # Assuming that the passed-in records are all United States data
+        if new_code is "national":
```
Comment (chinandrew) [inline _delphi_utils_python/delphi_utils/geomap.py:274]:
```suggestion
        if new_code == "national":
```

**Comment by krivard** [comment]
Merge is pending completion of [corresponding documentation](https://github.com/cmu-delphi/delphi-epidata/pull/231)

File: jhu/delphi_jhu/geo.py [hunk 2]
```diff
@@ -27,22 +27,18 @@ def geo_map(df: pd.DataFrame, geo_res: str):
...
-        df = gmpr.county_to_hrr(df, fips_col="fips", hrr_col="geo_id", date_col="timestamp")
-        df['geo_id'] = df['geo_id'].astype(int)
-    elif geo_res == 'county':
+    if geo_res == "fips":
         df.rename(columns={'fips': 'geo_id'}, inplace=True)
+    elif geo_res == "state":
+        df = gmpr.replace_geocode(df, "fips", "state_id", new_col="geo_id", date_col="timestamp")
+    else:
+        df = gmpr.replace_geocode(df, "fips", geo_res, new_col="geo_id", date_col="timestamp")
+
```
Comment (krivard) [inline jhu/delphi_jhu/geo.py:41]:
```suggestion

```
(remove trailing whitespace)

**Review by krivard — APPROVED** [review]
* delphi_utils lints and tests
* jhu lints (with python3.6; 3.7+ still has a pytest recursion problem) and tests
* emr_hosp is a retired signal, is exempt from linting, but tests pass
* verified no other signals are currently using GeoMapper

**Pull request merged**